app/javascript/components/VoteButton.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react';
+
+class VoteButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      liked: null,
+      like_id: null,
+      count: props.voteCount
+    }
+    this.recordLikes = this.recordLikes.bind(this)
+    this.selectedVote = this.selectedVote.bind(this)
+  }
+
+  componentDidMount(){
+    if (this.props.likes.length > 0) {
+      this.props.likes.forEach(voteRecord => {
+        if (voteRecord.user_id == this.props.currentUser) {
+          this.setState({liked: voteRecord.vote, like_id: voteRecord.id})
+        }
+      })
+
+    }
+  }
+
+  recordLikes(event) {
+    event.preventDefault();
+
```

Comment:
Since this is triggered by an `onClick`, you do not need `event.preventDefault()`

Reply:
Removed